src/storage/src/sst/parquet.rs

```diff
@@ -109,14 +115,91 @@ impl<'a> ParquetWriter<'a> {
                 .write(&arrow_batch)
                 .context(WriteParquetSnafu)?;
         }
-        arrow_writer.close().context(WriteParquetSnafu)?;
+
+        let file_meta = arrow_writer.close().context(WriteParquetSnafu)?;
+
+        let (start_timestamp, end_timestamp) =
+            match decode_timestamp_range(&file_meta, store_schema) {
+                Ok(Some((start, end))) => (Some(start), Some(end)),
+                Ok(None) => (None, None),
+                Err(e) => {
+                    error!(e;"Failed to calculate time range of parquet file");
+                    (None, None)
+                }
+            };
+
         object.write(buf).await.context(WriteObjectSnafu {
             path: object.path(),
         })?;
-        Ok(())
+        Ok(SstInfo {
+            start_timestamp,
+            end_timestamp,
+        })
     }
 }
 
+fn decode_timestamp_range(
+    file_meta: &FileMetaData,
+    store_schema: &StoreSchemaRef,
+) -> Result<Option<(Timestamp, Timestamp)>> {
+    let schema = store_schema.schema();
+    let (Some(ts_col_idx), Some(ts_col)) = (schema.timestamp_index(), schema.timestamp_column()) else { return Ok(None); };
+    let ts_datatype = &ts_col.data_type;
+    decode_timestamp_range_inner(file_meta, ts_col_idx, ts_datatype)
+}
+
+fn decode_timestamp_range_inner(
+    file_meta: &FileMetaData,
+    ts_index: usize,
+    ts_datatype: &ConcreteDataType,
+) -> Result<Option<(Timestamp, Timestamp)>> {
+    let mut start = i64::MAX;
+    let mut end = i64::MIN;
+
+    let unit = match ts_datatype {
+        ConcreteDataType::Int64(_) => TimeUnit::Millisecond,
+        ConcreteDataType::Timestamp(type_) => type_.unit(),
+        _ => {
+            return DecodeParquetTimeRangeSnafu {
+                msg: format!("Unexpected timestamp column datatype: {ts_datatype:?}"),
+            }
+            .fail();
+        }
+    };
+
+    for rg in &file_meta.row_groups {
+        let Some(ref metadata) = rg
+            .columns
+            .get(ts_index)
+            .context(DecodeParquetTimeRangeSnafu {
+                msg: format!("Cannot find ts column by index: {ts_index}"),
+            })?
+            .meta_data else { return Ok(None) };
+        let Some(stats) = &metadata.statistics else { return Ok(None) };
+        let (Some(min_value), Some(max_value)) = (&stats.min_value, &stats.max_value) else { return Ok(None); };
+
+        let min = i64::from_le_bytes(min_value[..8].try_into().map_err(|_| {
```

Comment:
is it always in little endian?

Reply:
I think we should log the error, or put it into `DecodeParquetTimeRange`'s msg.

Reply:
> is it always in little endian?

As per [parquet's implementation](https://docs.rs/parquet/29.0.0/src/parquet/file/statistics.rs.html#171-177), it should always be little endian.